src/itr_evaluator.ml

```diff
@@ -1347,64 +1347,81 @@ and evaluate_expr (context : 'a context) (e : expr) : record_item =
         { func = Hof { hof_type; lambda_args; body }; args = [ arg1; arg2 ] })
     ->
     let grab_record_list = function
-      | Rec_value (Value (Literal (Coll (_ct, args1)))) -> args1
+      | Rec_value (Value (Literal (Coll (_ct, args1)))) -> Ok args1
```

Comment:
May be clearer just returning an option here, rather than an extensible error in a result